evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -82,18 +90,27 @@ def validate(self, X, y):
                                                     details={"unsupported_type": y.logical_type.type_string}).to_dict())
         y_df = _convert_woodwork_types_wrapper(y.to_series())
         null_rows = y_df.isnull()
-        if null_rows.any():
+        if null_rows.all():
+            results["errors"].append(DataCheckError(message="Target values are either empty or fully null.",
+                                                    data_check_name=self.name,
+                                                    message_code=DataCheckMessageCode.TARGET_IS_EMPTY_OR_FULLY_NULL,
+                                                    details={}).to_dict())
+            return results
```

Comment:
Return here because I don't think it's beneficial to add other errors in this case (such as not having enough samples, etc) when this is the prime error!

Reply:
Right, if the user picks a fully null target, we tell them to go impute them. Once they have, they'll need to run the data checks again and could potentially run into the error of not having enough samples still, but I don't think it's super useful to tell them on their fully null target that they also might be missing enough samples of a specific class :')